### ISSUE TITLE
Clicking on "Return to site" on Stripe after the new onboarding flow results in a blocked account

### ISSUE THREAD

**Issue by daquinons** [Automattic/woocommerce-payments#6931]
### Describe the bug
After filling our new hosted onboarding form and being redirected to Stripe, if clicking on "Return to site" button, the Payments dashboard is displayed as is the account is onboarded, but both Payments and Deposits appear as restricted, with a callout to provide the missing details.

When clicking on the button to provide those missing details (meaning, finishing the KYC), the redirection fails and there is no obvious possibility to complete the onboarding.

See video and discussion here p1691127270624099-slack-C03KTTK2YMA

The user should be able to continue the onboarding with the already provided data. For that, we need to figure out why the redirection is not generated properly.

In case it's not possible to pick up from where it was left, we should take care of it in the background deleting the already existing account and prefilling a new one with the data that was already provided.

**Comment by daquinons** [comment]
Estimating for now with a 3, feel free to increase it after first investigation

**Comment by daquinons** [comment]
Putting it on hold until receive answer p1691478602206319-slack-C051WFFE961

**Comment by daquinons** [comment]
Forgot to add the update here, as the issue is been unblocked since last week: we need to use "account link" instead of "login link" for those accounts that are not fully onboarded yet.

**Comment by hobhobuk** [comment]
This issue still very much exists.  Having to contact support is not a fix.
